### DISCUSSION TITLE
Adding write support to the SD card and other MMC things

### DISCUSSION THREAD

**Discussion by Wolfvak** [linux-3ds/arm9linuxfw#19] [General]
should SD write support be enabled? it's only a few lines afaict and performance isn't too bad, but I'm afraid someone will end up messing the data in their card and blame us because "it's unstable and shouldn't be released yet" or something like that

even I'm somewhat afraid of toggling the wrong bit at the wrong time and having the whole FS corrupt itself

adding NAND support is also possible but out of the question right now - it's too dangerous and _will_ end up bricking someone

if anyone knows any virtio people, maybe we should suggest adding a MMC device type? they have things that make them stand out from plain "block devices", like clock controls, sleep mode, switching between 1 and 4 bit mode, multiple voltage select, internal registers like the CID/CSD and other funky things that can cause the card to die (or be revived, if you have a locked NAND it _could_ be unlocked directly from linux)

any thoughts?

**Comment by FreddyFeuerstein** [comment]
Maybe make a new branch for it and put 'still in development. use at own risk' in the readme of that branch. And a suggestion to make frequent backups and use a fresh card for that fork.

also, I wouldn't mess with the nand bc its too easy to mess it up and loose all your progress in all your games.

**Reply by Wolfvak** [reply]
alright, I might try doing that and delay merging it to master until I feel it's stable enough

**Reply by iAmInActions** [reply]
nice. I could test it for you if you don't want to use your 3ds for it. I have spare SD cards and know how to reflash the nand so if anything goes wrong, it won't be a problem for me.

**Reply by FreddyFeuerstein** [reply]
man I really need this. maybe i could try to do it myself but i don't even know where the adress of the device is.

**Comment by iAmInActions** [comment]
I really need write support.
First of all, the x server needs to create a lock file to work and won't run when it can't.
Second thing is, my space invaders high score cannot be saved.
Third thing why I need it is doom save states.
